test/src/e2e_vm_tests/test_programs/should_pass/language/storage_slot_sized/src/main.sw

```diff
@@ -0,0 +1,77 @@
+contract;
+
+struct S1 {
+  f1: u64,
+  f2: u64,
+  f3: u64,
+  f4: u64,
+}
+
+impl PartialEq for S1 {
+  fn eq(self, other: Self) -> bool {
+      self.f1 == other.f1 && self.f2 == other.f2 && self.f3 == other.f3 && self.f4 == other.f4
+  }
+}
+
+impl Eq for S1 {}
+
+storage {
+    var1: S1 = S1 { f1: 1, f2: 2, f3: 3, f4: 4 },
+    var2: (S1, S1) = (S1 { f1: 11, f2: 21, f3: 31, f4: 41 }, S1 { f1: 12, f2: 22, f3: 32, f4: 42 })
+}
+
+abi StorageTest {
```

Comment:
There is actually no need for `abi` anymore. It's just annoying to keep it in sync with the `Contract` and it bloats the test. It is enough to have `impl Contract`. One of my favorite language features exactly because of tests 😄